client/session.go

```diff
@@ -717,13 +716,14 @@ func (s *session) setTopologyWithLock(topoMap topology.Map, queues []hostQueue,
 	s.pools.fetchBatchOpArrayArray.Init()
 
 	if s.pools.iteratorArray == nil {
-		s.pools.iteratorArray = encoding.NewIteratorArrayPool([]pool.Bucket{
-			pool.Bucket{
-				Capacity: replicas,
-				Count:    s.opts.SeriesIteratorPoolSize(),
-			},
-		})
-		s.pools.iteratorArray.Init()
+		size := replicas * s.opts.SeriesIteratorPoolSize()
+		poolOpts := pool.NewObjectPoolOptions().
+			SetSize(size).
+			SetInstrumentOptions(s.opts.InstrumentOptions().SetMetricsScope(
+				s.scope.SubScope("iterator-pool"),
+			))
+		s.pools.iteratorArray = encoding.NewMultiReaderIteratorPool(poolOpts)
```

Comment:
don't think you need this, `s.pools.multiReaderIterator` already has the type you're trying to migrate to